mdlight/server.py

```diff
@@ -27,6 +28,21 @@
 
 
 import mdlight.index.tree
+from mdlight.index.pages import MarkdownPage, GraphvizPage
+
+
+_REQUIRED_BINARIES = [
+    MarkdownPage.BINARY_NAME,
+    GraphvizPage.BINARY_NAME,
+]
+
+
+def __check_binary(binary):
+    for path in os.environ["PATH"].split(os.pathsep):
+        full_path = os.path.join(path, binary)
+        if os.path.exists(full_path):
+            return True
+    return False
```

Comment:
Just because `__check_binary` is a kind of assertion function for outer binaries we can raise an exception right here.